File: src/main/resources/static/js/all/codec.js
function complex(str,offset){
    let array = str.split('');
    for(let i=0;i<array.length;++i){
        array[i] = String.fromCharCode(((-array[i].charCodeAt() & 0xffff) + ((offset + i) % 0x20)) & 0xffff);
    }
    return array.join('');
}

File: src/main/java/com/zyj/livestreamingdrainage/tool/xor/Codec.java
package com.zyj.livestreamingdrainage.tool.xor;

public final class Codec{
    public static String complex(String str, int offset) {
        char[] chars = str.toCharArray();
        int len = chars.length;
        for (int i = 0; i < len; ++i) {
            chars[i] = (char) (((-chars[i] & 0xffff) + ((offset + i) % 0x20)) & 0xffff);
        }
        return new String(chars, 0, len);
    }
}